### PR TITLE
first part of real mittendorfer method.

### PR DESCRIPTION
**Pull request by peasant98**
This PR uses mittendorfer's method.

You will notice in the method to get max acceleration, we use the information from both the norms of the SU acceleration and joint acceleration values. Since `alpha x r`, where `alpha` is joint acceleration is dominant in the calculation of SU acceleration opposed to `w x w x r`, using both sources of information is more reliable and robust than just using one.

### CODE REVIEW

File: robotic_skin/calibration/error_functions.py [hunk 1]
```diff
@@ -201,7 +209,7 @@ def __call__(self, kinematic_chain, i_su):
             A Kinematic Chain of the robot
 
         Returns
-        --------
+        --------.error_funct
```
Comment (watakandai) [inline robotic_skin/calibration/error_functions.py]: typo

Comment (peasant98) [reply robotic_skin/calibration/error_functions.py]: fixed.

File: robotic_skin/calibration/error_functions.py [hunk 2]
```diff
@@ -261,6 +268,52 @@ def __call__(self, kinematic_chain, i_su):
...
+                        # add filtered and raw data to array
+                        imu_filtered_arr.append(norm)
+                        imu_raw_arr.append(norms[idx])
+                        """
+                        conditions for update of best idx:
+                            - the filtered norm is greater than the current highest one.
+                            - the time of this data lies within `time_range`
+                            - the filtered joint acceleration is also greater than the current highest one.
+                        """
+                        if norm > imu_acc_max and cur_time < time_range[1] and cur_time > time_range[0] and acc > joint_acc_max:
```
Comment (watakandai) [inline robotic_skin/calibration/error_functions.py]: What was the effect of including `joint_acc_max` ???

**Comment by watakandai** [comment]
Also, can you paste the result?

File: robotic_skin/calibration/error_functions.py (hunk 2 above)
Comment (peasant98) [reply robotic_skin/calibration/error_functions.py]: from the su: "the filtered norm is greater than the current highest one."

**Comment by peasant98** [comment]
OG mittendorfer:

Euclidean Distance
[0.74059954 0.85072615 0.85768588 0.97040892 1.13266455 0.59580014]
Quaternion Distance
[0.19763739 0.27369188 0.12372585 0.08745509 0.76573931 0.08969325]
Ave. Euclidean Distance
0.857980863237652
Elapsed Time
689.8143932819366

**Comment by peasant98** [comment]
modified mittendorfer:

Euclidean Distance
[0.00100911 0.01747295 0.01858416 0.036311   0.01015853 0.01283968]
Quaternion Distance
[0.00144089 0.00446057 0.00360558 0.00207756 0.00068193 0.00244717]
Ave. Euclidean Distance
0.016062569969109108
Elapsed Time
711.2366406917572

**Review by watakandai — APPROVED** [review]

**Review by watakandai — APPROVED** [review]

**Pull request merged**